src/send_future.rs

```diff
@@ -62,21 +54,53 @@ impl<R, F> SendFuture<R, F, ResolveToHandlerReturn> {
     }
 }
 
-impl<R> SendFuture<R, BoxFuture<'static, Receiver<R>>, ResolveToHandlerReturn> {
-    pub(crate) fn sending_boxed<F>(send_fut: F) -> Self
+pub(crate) mod private {
+    pub trait SetPriority {
+        fn set_priority(&mut self, priority: u32);
+    }
+}
+
+impl<R, F, TResolveMarker> SendFuture<R, F, TResolveMarker>
+where
+    F: SetPriority,
+{
+    /// Set the priority of a given message. See [`Address`] documentation for more info.
+    ///
+    /// Panics if this future has already been polled.
+    pub fn priority(mut self, new_priority: u32) -> Self {
+        match &mut self.inner {
+            SendFutureInner::Sending(inner) => inner.set_priority(new_priority),
+            SendFutureInner::Receiving(_) => {
+                panic!("setting priority after polling is unsupported")
+            }
+            SendFutureInner::Done => panic!("setting priority after polling is unsupported"),
+        }
+
+        SendFuture {
+            inner: self.inner,
+            phantom: PhantomData,
+        }
+    }
+}
+
+impl<R> SendFuture<R, ActorErasedSending<R>, ResolveToHandlerReturn> {
+    pub(crate) fn sending_boxed<F>(sending: F, rx: catty::Receiver<R>) -> Self
```

Comment:
```suggestion
    pub(crate) fn sending_erased<F>(sending: F, rx: catty::Receiver<R>) -> Self
```

Perhaps?

Reply:
Sounds good. I won't commit the suggestion just since it will entail other renamings